workers/ohsome_quality_analyst/config/config.py

```diff
@@ -0,0 +1,168 @@
+"""Load configuration from environment variables or configuration file on disk."""
+
+import errno
+import logging
+import logging.config
+import os
+import sys
+
+import rpy2.rinterface_lib.callbacks
+import yaml
+
+from ohsome_quality_analyst import __version__ as oqt_version
+
+
+def get_config_path():
+    """Get configuration file path from environment variable 'OQT_CONFIG'.
+
+    Defaults to 'config.yaml'
+    """
+    return os.getenv(
+        "OQT_CONFIG",
+        default=os.path.join(
+            os.path.dirname(
+                os.path.abspath(__file__),
+            ),
+            "config.yaml",
+        ),
+    )
+
+
+def load_config_from_file(path: str) -> dict:
+    """Load configuration from file on disk.
+
+    Remove items where value is None (`null` in YAML)
+    """
+    with open(path, "r") as f:
+        cfg = yaml.safe_load(f)
+    return {k: v for k, v in cfg.items() if v is not None}
+
+
+def load_config_from_env() -> dict:
+    """Load configuration from environment variables."""
+    cfg = {
+        "postgres_host": os.getenv("POSTGRES_HOST"),
+        "postgres_port": os.getenv("POSTGRES_PORT"),
+        "postgres_db": os.getenv("POSTGRES_DB"),
+        "postgres_user": os.getenv("POSTGRES_USER"),
+        "postgres_password": os.getenv("POSTGRES_PASSWORD"),
+        "data_dir": os.getenv("OQT_DATA_DIR"),
+        "geom_size_limit": os.getenv("OQT_GEOM_SIZE_LIMIT"),
+        "ohsome_api": os.getenv("OQT_OHSOME_API"),
+        "user_agent": os.getenv("OQT_USER_AGENT"),
+    }
+    return {k: v for k, v in cfg.items() if v is not None}
+
+
+def get_config() -> dict:
+    """Get configuration variables from environment and file.
+
+    Configuration variables from environment will be given precedence.
+    """
+    path = get_config_path()
+    cfg = load_config_from_file(path)
+    # Give precedence to environment variables
+    cfg.update(load_config_from_env())
+    # Set default values
+    if "datasets" not in cfg.keys():
+        cfg["datasets"] = {
+            "regions": {
+                "default": "ogc_fid",
+                "other": ["name"],
+            }
+        }
+    if "data_dir" not in cfg.keys():
+        cfg["data_dir"] = get_default_data_dir()
+    if "user_agent" not in cfg.keys():
+        cfg["user_agent"] = "ohsome-quality-analyst/{}".format(oqt_version)
+    cfg = check_missing_conf_keys(cfg)
+    return cfg
+
+
+def check_missing_conf_keys(cfg) -> dict:
+    """Check for missing configuration keys.
+
+    Returns:
+        dict: Input dict without changes
+    Raises:
+        KeyError: In case configuration key is missing from configuration dictionary.
+    """
```

Comment:
What is the semantic difference between a missing key and a key explicitly set to `None`/`none`? I would say there isn't any and therefore `None`-keys should be ignored.

Reply:
This validation function has been removed in favor of default values if config is not provided.